examples/CMakeLists.txt

```diff
@@ -1,59 +1,67 @@
 cmake_minimum_required(VERSION 3.15)
 project (crow_examples)
 
+# Define Required libraries 
+list(APPEND REQUIRED_LIBRARIES 
+  ${Boost_LIBRARIES}
+  ${CMAKE_THREAD_LIBS_INIT}
+)
+
 if (MSVC)
   add_executable(example_vs example_vs.cpp)
-  target_link_libraries(example_vs ${Boost_LIBRARIES})
-  target_link_libraries(example_vs ${CMAKE_THREAD_LIBS_INIT})
+  target_compile_options(example_vs PRIVATE "${compiler_options}")
+  target_link_libraries(example_vs )
 else ()
   add_executable(helloworld helloworld.cpp)
-  target_link_libraries(helloworld ${Boost_LIBRARIES})
-  target_link_libraries(helloworld ${CMAKE_THREAD_LIBS_INIT})
+  target_compile_options(helloworld PRIVATE "${compiler_options}")
+  target_link_libraries(helloworld PUBLIC ${REQUIRED_LIBRARIES})
 
-if (OPENSSL_FOUND)
-  add_executable(example_ssl ssl/example_ssl.cpp)
-  target_link_libraries(example_ssl ${Boost_LIBRARIES})
-  target_link_libraries(example_ssl ${CMAKE_THREAD_LIBS_INIT} ${OPENSSL_LIBRARIES})
-endif()
+  if (OPENSSL_FOUND)
+    add_executable(example_ssl ssl/example_ssl.cpp)
+    target_compile_options(example_ssl PRIVATE "${compiler_options}")
+    target_link_libraries(example_ssl PUBLIC ${REQUIRED_LIBRARIES} ${OPENSSL_LIBRARIES})
+  endif()
```

Comment:
Maybe add info that example wasn't build when there is no SSL lib?

Reply:
Sorry for being imprecise - by 'adding message' I've meant message displayed by CMake, so that person building examples would know that this one hasn't been built.

Reply:
I see, it was not your fault, I've just interpreted it in another way. Sure, I'll add a status message in case of OpenSSL missing